Library/Homebrew/rubocops/python_internal.rb

```diff
@@ -0,0 +1,31 @@
+# typed: true
+# frozen_string_literal: true
+
+require "rubocops/extend/formula"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      # This cop audits `python-internal` dependencies in formulae.
+      class PythonInternalCop < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, _body_node)
+          dep = depends_on?("python-internal")
+
+          return if dep.blank?
+
+          unless tap_style_exception? :python_internal_allowlist
+            problem "This formula is not allowed to depend on python-internal."
+          end
+
+          dep.child_nodes.each do |child_node|
+            next if !child_node.instance_of?(RuboCop::AST::StrNode) &&
+                    child_node.values.first.child_nodes.length.zero? &&
+                    child_node.values.first.value == :build
+
+            problem "The python-internal dependency can only be a :build dependency."
+          end
```

Comment:
An alternative to this is to use the `def_node_matcher` tool.

Outside of the `audit_formula` method you can call:

```ruby
def_node_matcher :build_only_dependency?, <<~PATTERN
  (send nil? :depends_on (hash (pair str (sym :build))))
PATTERN
```

Then use:

```suggestion
          return if build_only_dependency? dep

          problem "The python-internal dependency can only be a :build dependency."
```

In my opinion, this is a little cleaner and easier to understand. All it's doing is saying "make sure the `depends_on` line matches this exact pattern of a build-only dependency`. This isn't necessary, though, so using the current method works just fine if that's what's preferred.

Reply:
I used the other solution, but I have no preference.